### PR TITLE
New connection handling

### PR DESCRIPTION
**Pull request by l-jonas**
Goals:

- code which is easier to read/ maintain
- connecting faster

### CODE REVIEW

File: syncthing-bep/src/main/kotlin/net/syncthing/java/bep/connectionactor/OpenConnection.kt [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+                logger.debug("opening relay connection")
+                keystoreHandler.wrapSocket(RelayClient(configuration).openRelayConnection(address), KeystoreHandler.BEP)
+            }
+            DeviceAddress.AddressType.HTTP_RELAY, DeviceAddress.AddressType.HTTPS_RELAY -> {
+                logger.debug("opening http relay connection")
+
+                throw IOException("http relay support is disabled")
+                // This is not documented anywhere (https://forum.syncthing.net/t/http-relays/12367) so it's disabled
+                // it's still kept because it is eventually useful in the future
+                // keystoreHandler.wrapSocket(HttpRelayClient().openRelayConnection(address), KeystoreHandler.BEP)
```
Comment (Nutomic) [inline syncthing-bep/src/main/kotlin/net/syncthing/java/bep/connectionactor/OpenConnection.kt]: I would prefer if you just remove the useless code (anything related to http relay). If it's ever needed, we can get it from the git history. Right now it's just noise. And we also don't need backward compatibility in the library.

**Comment by Nutomic** [comment]
Wow this is a huge PR, and I don't really have the time to review all of it. If it works better for you, I'm happy to merge it.

By the way, unrelated to this PR but Travis is showing a lot of warnings in the build log. Would be good to fix them, or suppress irrelevant ones. Running Analyze -> Inspect Code in Android Studio should actually fix most of them, but that's not working right for me...

**Comment by l-jonas** [comment]
@Nutomic 

> If it works better for you

I would have to to do some tests to check if it works better than the current implementation (I only tested that it works with <https://github.com/syncthing/syncthing-lite/pull/71/commits/9b95e589094a103ba5363b4e472bbdd4c1bae087>). However, the new code is easier to understand for me and it is optimized in some parts so that it takes theoretically less time until a connection is established.

> I don't really have the time to review all of it

It would be no problem if you would need 1 or 2 weeks. But if you have got really no time, then I would review it myself in a few days and test it again.

**Comment by Nutomic** [comment]
I should have said "if it's better than the previous code", and if it's easier to understand and better optimized, that's definitely worth merging.

I will try to review it, but no guarantees.

**Comment by l-jonas** [comment]
New shared folders require a restart of the App with this. This makes the setup wizard hang at the last step.

**Pull request merged**